rfcs/rfc-business-ledger-data-structure.md

```diff
@@ -0,0 +1,83 @@
+- Start Date: 2021-04-22
+- Target Major Version: [information on a ledger](https://github.com/bcgov/entity/issues/6969)
+- Reference Issues: n/a
+- Entity Issue: github/entity#7193
+- Implementation PR: (leave this empty)
+
+# Summary
+
+Bulk listings of filings should return a catalog of filings, with enough information to be useful, to allow the user to further select the filing of interest.
+
+The filing endpoint should not default to return the complete filing, of every filing on the businesses ledger.
+
+# Basic example
+
+The RESTful style call to ```GET /businesses/{business_identifier}/filings``` will return a cata log of filings.
+
+The RESTful style call to ```GET /businesses/{business_identifier}/filings/{filing_id}``` will return the full individual filing.
+
+The catalog call follows the following pattern:
+``` json
+{"filings":[
+    {"name": "annualReport",
+     "status": "Completed",
+     "effectiveDate": "2003-07-01T00:00:00+00:00",
+     "availableOnPaperOnly": True,
+     ...
+     "filingLink": "https://api.bcregistry.ca/v1/businesses/CP00001/filings/1243",
+     ...
+    },
+]}
+```
+
+# Motivation
+
+The Business Ledger should be a quick and useful access point into managing your business.
+Most of the time people are accessing their business to update it, file an Annual Report, and maybe print off a historical filing or receipt.
+Creating and returning all of the filings, to show almost none of the filing data a company has ever submitted to the registrar is neither quick, efficient or meets the needs of the customer base.
+
+For note, most corporations should have copies of their filings in the corporate records that they are to maintain at their registered office.
+
+
+# Detailed design
+
+This RFC is only to create the data structure for the catalog of filings.
+The first release of the data structure will have the following elements.
+``` json
+{"filings":[
+    {
```

Comment:
I will scan the existing filing history list to see what other fields we might want in here (tomorrow)... unless you have already done it? I have a hunch this list is too short.

Reply:
See the **Unresolved questions**.
I did ask what needed to be displayed per filing. If there are fields, we can follow a similar take to Cloud-Events and have an extra data field that has meaning based on the event, in our case filing.

Reply:
I'm interrupted by a high priority ticket atm. I've asked for help. Maybe someone else can look at the Filings UI code and look for missing fields. If not, I assume we can add them as we go?

Reply:
@severinbeauvais Will have a look 👍 

Curious do we have some designs we should reference for future state or are we only worried about current state of the filings history list?